provider/cluster/kube/client_exec.go

```diff
@@ -53,52 +55,19 @@ func (c *client) Exec(ctx context.Context, leaseID mtypes.LeaseID, serviceName s
 	tsq remotecommand.TerminalSizeQueue) (ctypes.ExecResult, error) {
 	namespace := builder.LidNS(leaseID)
 
-	// Check that the deployment exists
-	deployments, err := c.kc.AppsV1().Deployments(namespace).List(ctx, metav1.ListOptions{
-		TypeMeta:             metav1.TypeMeta{},
-		LabelSelector:        "",
-		FieldSelector:        "",
-		Watch:                false,
-		AllowWatchBookmarks:  false,
-		ResourceVersion:      "",
-		ResourceVersionMatch: "",
-		TimeoutSeconds:       nil,
-		Limit:                0,
-		Continue:             "",
-	})
+	_, err := c.kc.CoreV1().Services(namespace).Get(ctx, serviceName, metav1.GetOptions{})
```

Comment:
We can just drop this query & check, since the service is basically always going to exist. If the pod below isn't present I think an appropriate error message comes back already.

Reply:
removed